probe-rs/src/architecture/arm/core/armv8m.rs

```diff
@@ -160,16 +163,16 @@ impl<'probe> CoreInterface for Armv8m<'probe> {
         let _ = self.status()?;
 
         const XPSR_THUMB: u32 = 1 << 24;
-        let xpsr_value: u32 = self.read_core_reg(register::XPSR.id)?.try_into()?;
+        let xpsr_value: u32 = self.read_core_reg(self.registers().psr()?.id)?.try_into()?;
         if xpsr_value & XPSR_THUMB == 0 {
-            self.write_core_reg(register::XPSR.id, (xpsr_value | XPSR_THUMB).into())?;
+            self.write_core_reg(self.registers().psr()?.id, (xpsr_value | XPSR_THUMB).into())?;
```

Comment:
The problem with this API now is that the PSR could not exists but we still use it hardcoded in code here.
We are using a fallible/dynamic API in static/infallible code. I wonder if we could abstract this better.

Reply:
@Yatekii ... this is probably my lack of understanding, but even in the original code, the entire line `let xpsr_value` is fallible in more than one place. The only other places where `psr()` is called, it was already being used behind either a `is_some()` which is now an `is_ok()` or behind a similar `match`. 

That said, I see your point, so if you are able to suggest something cleaner, I'd be happy to implement it :)

Reply:
... just thinking about it some more. Should I make methods for some of the registers like `psr()` return an `Option<>` instead of a `Result<>`, or ???

I should also point out, that for mandatory registers like `pc`, `sp`, etc. I already check for their presence when `RegisterFile::new()` is called, so any bugs in `RegisterFile` contents where those are concerned, will be flagged very early in the code execution.

Reply:
@Yatekii ... this is probably my lack of understanding, but even in the original code, the entire line `let xpsr_value` is fallible in more than one place. The only other places where `psr()` is called, it was already being used behind either a `is_some()` which is now an `is_ok()` or behind a similar `match`. 

That said, I see your point, so if you are able to suggest something cleaner, I'd be happy to implement it :)